src/features/Sync/SyncQueueManager.ts

```diff
@@ -0,0 +1,58 @@
+import { DatabaseService } from '../../common/services';
+import { SyncLog } from '../../common/services/Database/entities';
+
+const SYNC_QUEUE_PEEK_NEXT = `
+    SELECT * FROM synclog 
+    ORDER BY timestamp
+    LIMIT ?
+`;
+
+const SYNC_QUEUE_PEEK_ALL = `
+    SELECT * FROM synclog 
+    ORDER BY timestamp
+`;
+
+const SYNC_QUEUE_DROP_NEXT = `
+  DELETE FROM synclog
+  WHERE synclog.id IN 
+  (
+    SELECT id
+    FROM synclog
+    ORDER BY timestamp
+    LIMIT ?
+  ) 
+`
+
+const SYNC_QUEUE_DROP_ALL = `
+  DELETE FROM synclog
+`
+
+const SYNC_QUEUE_LENGTH = `
+  SELECT COUNT(*) FROM synclog
+`
+
+class SyncQueueManager {
```

Comment:
The basic `SyncQueue` data structure. Pretty much just a queue (FIFO based on timestamp).